scipy/io/tests/test_mmio.py

```diff
@@ -111,6 +127,137 @@ def test_random_rect_real(self):
         b = mmread(fn)
         assert_array_almost_equal(a,b)
 
+
+class TestMMIOSparseCSR(TestMMIOArray):
+
+    def setUp(self):
+        self.tmpdir = mkdtemp()
+        self.fn = os.path.join(self.tmpdir, 'testfile.mtx')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmpdir)
+    
+    def test_simple(self):
+        a = [[1,2],[3,4]]
```

Comment:
If you prefer to do something like:

```
# Need to avoid `test` in the method name
def check_single(a, info):
    mmwrite(self.fn, a)
    assert_equal(mminfo(self.fn), info)
    b = mmread(self.fn)
    assert_array_almost_equal(a, b)
```

then

```
    def test_simple_integer(self):
        check_single(scipy.sparse.csr_matrix([[1, 2], [3, 4]]), (2, 2,
4, 'array', 'integer', 'general'))
```

that also gets rid of the code duplication, and is as readable.